static/app/utils/url/stripURLOrigin.tsx

```diff
@@ -0,0 +1,16 @@
+import {safeURL} from 'sentry/utils/url/safeURL';
+
+/**
+ * Accept a url like:
+ * `https://example.com/path/name?query=params#hash` and return
+ * `/path/name?query=params#hash`
+ */
+export default function stripURLOrigin(url: string): string {
+  const parsedUrl = safeURL(url);
+
+  if (!parsedUrl) {
+    return url;
+  }
+
+  return parsedUrl.pathname + parsedUrl.search + parsedUrl.hash;
```

Comment:
This is wrong. A url's hash can be `#` or simply ``. If we look into the Node.js implementation,


https://github.com/nodejs/node/blob/main/lib/internal/url.js#L1030

We currently lack the context of `hasHash` attribute.


If we look into the URL Spec (Ref; https://url.spec.whatwg.org/#dom-url-hash)

> If [this](https://webidl.spec.whatwg.org/#this)’s [URL](https://url.spec.whatwg.org/#concept-url-url)’s [fragment](https://url.spec.whatwg.org/#concept-url-fragment) is either null or the empty string, then return the empty string.


For example, 

`new URL('https://google.com')` and `new URL(`'https://google.com#')` returns empty string for hash, but it includes `#` in `.href` attribute.

```
> new URL('https://www.google.com')
URL {
  href: 'https://www.google.com/',
  origin: 'https://www.google.com',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'www.google.com',
  hostname: 'www.google.com',
  port: '',
  pathname: '/',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
> new URL('https://www.google.com#')
URL {
  href: 'https://www.google.com/#',
  origin: 'https://www.google.com',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'www.google.com',
  hostname: 'www.google.com',
  port: '',
  pathname: '/',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
```

Reply:
Damn, that is sneaky. I thought this would still return # for URL.hash... I'll revert the change

Reply:
Updated and left a comment referencing your explanation 🙏🏼